### PR TITLE
sys/log: Add pretty printing for CBOR log entries

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**
...and also some useful additions to CLI.

### CODE REVIEW

File: sys/log/full/src/log_shell.c [hunk 1]
```diff
@@ -99,6 +242,15 @@ shell_log_dump_all_cmd(int argc, char **argv)
             continue;
         }
 
+        if (list_only) {
+            console_printf("%s\n", log->l_name);
+            continue;
+        }
+
+        if ((argc > 1) &&  strcmp(log->l_name, argv[1])) {
```
Comment (kasjer) [inline sys/log/full/src/log_shell.c]: extra space after &&

Comment (kasjer) [reply sys/log/full/src/log_shell.c]: searching for next log could stop after one that was requested was dumped

File: sys/log/full/src/log_shell.c [hunk 2]
```diff
@@ -45,49 +167,70 @@ shell_log_dump_entry(struct log *log, struct log_offset *log_offset,
...
 #else
     switch (ueh->ue_etype) {
     case LOG_ETYPE_STRING:
-        console_printf("[%llu] %s\n", ueh->ue_ts, data);
+        console_write(data, strlen(data));
+        break;
+    case LOG_ETYPE_CBOR:
+        log_shell_cbor_reader_init(&cbor_reader, log, dptr, len);
+        cbor_parser_init(&cbor_reader.r, 0, &cbor_parser, &cbor_value);
+        cbor_value_to_pretty(stdout, &cbor_value);
```
Comment (kasjer) [inline sys/log/full/src/log_shell.c]: printing to stdout would end up on console as other output, right?

File: sys/log/full/src/log_shell.c (hunk 1 above)
Comment (andrzej-kaczmarek) [reply sys/log/full/src/log_shell.c]: agree, will fix

File: sys/log/full/src/log_shell.c (hunk 2 above)
Comment (andrzej-kaczmarek) [reply sys/log/full/src/log_shell.c]: correct

File: sys/log/full/src/log_shell.c [hunk 3]
```diff
@@ -36,6 +36,128 @@
...
+                          int src_offset, size_t len)
+{
+    struct log_shell_cbor_reader *cbr = (struct log_shell_cbor_reader *)d;
+    uint8_t buf[16];
+    int chunk_len;
+    int offset;
+
+    offset = 0;
+
+    while (offset < len) {
```
[the remark this answers is not in the document]
Comment (andrzej-kaczmarek) [reply sys/log/full/src/log_shell.c]: I do not know in advance how long the data is and also do not want to allocate huge buffers on stack. perhaps current buffer can be made a bit larger though, not sure exactly where this is used in tinycbor

[the remark this answers is not in the document]
Comment (andrzej-kaczmarek) [reply sys/log/full/src/log_shell.c]: ah, you're right - reading in chunks makes sense for memcmp but for memcpy we can read at once :-)

**Pull request merged**